_posts/2021-09-02-gsoc-final-report-orion.md

```diff
@@ -0,0 +1,195 @@
+
+At the outset of my [GSoC project], I hoped to create a package with:
+
+- Functions for selecting the solvation shells of ions.
+- Automated analysis of ion coordination environment.
+- Documentation, testing, and well-written code.
+
+After 10 weeks of working with the [MDAnalysis] team, the [solvation_analysis] package vastly exceeds that original intent. There is a robust and simple API, the functionality generalizes to non-ionic systems, solvation distances are automatically identified, and it's <b>really fast.</b> Further, I've learned a huge amount about software development, the package release cycle, and the world of open-source.
+
+In this post, I will summarize my contributions to `solvation_analysis` and several features of the package.
+
+## My Summer in Pull Requests
+
+### Template for a snazzy package
+PR [#7] and [#20]
+
+My summer began with a duo of pull requests. PRs #7 and #20 filled out the template generated by the [MolSSI cookiecutter], creating a fashionable scaffold that my code could fit into. In essence, the PRs set up the trappings of a real package, assuring users (and myself) of a certain core competence in the development.
+
+### Testing, testing, one-two-three
+PR [#6], [#16], and [#17]
+
+Robust testing is part of a finished package. PR #6 introduced the initial code base and some rudimentary testing. In PR #16, I started using [pytest], which made writing and using tests much easier. Finally, PR #17, integrated testing with GitHub CI, finalizing a critical component of my development workflow. 
+
+### Cutting off radial distribution functions
+PR [#25]
+
+PR #25 was a science challenge in addition to a software challenge. I needed to robustly identify the first minima of noisy RDF data so I could automate the solvation analysis. My approach was to perform a polynomial interpolation of the RDF and then take the 2nd-derivative to find the minima. This worked well for smooth RDFs but failed for RDFs with noise. As a quick fix, I added in several heuristics to get the solvation distances identified on my noisy test data. For now, the RDF parser works reasonably well for smooth data and warns users when it fails. An interpolation of an RDF is shown below. 
+
+![interpolation]({{site.images}}/final_report_orion/interpolation.png)
+
+### Analysis: the mega-PR
+PR [#29]
+
+PR #29 is responsible for the majority of functionality in solvation_analysis. It implemented the Solution class and all of the analysis functions, reaching 250 comments from myself and my mentors. Much of the discourse focused on iterating through API designs, which eventually led to a very clean Solution-centered package. The underlying data structure is a simple `Pandas.DataFrame` that allows all analyses to be vectorized. As shown below, the `DataFrame` is indexed by step in the analysis (`frame`), the `solvated_atom` from the solute, and the solvent atoms `atom_id`. This provides a unique identifier for each atom participating in solvation.
```

Comment:
If this is a "tidy dataframe" (or long dataframe) then perhaps link to https://tidyr.tidyverse.org/articles/tidy-data.html and the tidy data paper by Hadley Wickham — data science people might appreciate the thought that went into your design

Reply:
I linked to the article, thanks for the suggestion!